spackle-android-lib/src/main/kotlin/co/electriccoin/zcash/spackle/Twig.kt

```diff
@@ -0,0 +1,81 @@
+package co.electriccoin.zcash.spackle
+
+import android.content.Context
+import android.util.Log
+import co.electriccoin.zcash.spackle.process.ProcessNameCompat
+import java.util.Locale
+
+/**
+ * A twig is a tiny log. These logs are intended for development rather than for high performance
+ * or usage in production.
+ */
+object Twig {
+    /**
+     * Format string for log messages.
+     *
+     * The format is: <Process> <Thread> <Class>.<method>(): <message>
+     */
+    private const val FORMAT = "%-33s %-30s %s.%s(): %s" // $NON-NLS-1$
+
+    @Volatile
+    private var tag: String = "Twig"
+
+    @Volatile
+    private var processName: String = ""
+
+    /**
+     *  For best results, call this method before trying to log messages.
+     */
+    fun initialize(context: Context) {
+        tag = getApplicationName(context)
+        processName = ProcessNameCompat.getProcessName(context)
+    }
+
+    fun info(message: () -> String) {
+        Log.i(tag, formatMessage(message))
+    }
+
+    fun warn(message: () -> String) {
+        Log.w(tag, formatMessage(message))
+    }
+
+    private fun formatMessage(message: () -> String): String {
+        val currentThread = Thread.currentThread()
+        val trace = currentThread.stackTrace
+        val sourceClass = trace[2].className
+        val sourceMethod = trace[2].methodName
+
+        return String.format(
+            Locale.ROOT,
+            FORMAT,
+            processName,
+            currentThread.name,
+            cleanupClassName(sourceClass),
+            sourceMethod,
+            message()
+        )
+    }
+}
+
+/**
+ * Gets the name of the application or the package name if the application has no name.
+ *
+ * @param context Application context.
+ * @return Label of the application from the Android Manifest or the package name if no label
+ * was set.
+ */
+fun getApplicationName(context: Context): String {
+    val applicationLabel = context.packageManager.getApplicationLabel(context.applicationInfo)
+
+    return applicationLabel.toString().lowercase(Locale.ROOT).replace(" ", "-")
+}
+
+private fun cleanupClassName(classNameString: String): String {
```

Comment:
I still need to test the behavior here with lambdas, top level methods, and inline methods.